src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -220,11 +220,12 @@ private ResponseEntity<GatewayCallbackResponse> handleIllegalDataStateException(
    * @param hipNotifyRequest Has careContexts details for dataTransfer
    */
   @PostMapping({"/v0.5/consents/hip/notify"})
-  public void initiateDataOnNotify(@RequestBody HIPNotifyRequest hipNotifyRequest)
+  public ResponseEntity<GatewayCallbackResponse> initiateDataOnNotify(@RequestBody HIPNotifyRequest hipNotifyRequest)
       throws IllegalDataStateException {
     if (hipNotifyRequest != null) {
       workflowManager.initiateConsentOnNotify(hipNotifyRequest);
     } else log.debug("Error in response of Consent Notify");
```

Comment:
Since its an error in the else block, you could return a BAD_REQUEST back to the client.